cpp/include/cudf/io/detail/experimental/hybrid_scan.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
```

Comment:
Move contents to the public header and delete this one, as `hybrid_scan_reader` cannot be constructed and returned by API 1 with just a forward declaration.